code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/NotificationBuilder.kt

```diff
@@ -67,9 +83,175 @@ object NotificationBuilder {
+
+        // calculate difference in fire date from the current date if timestamp is provided
+        val secondsUntilFireDate: Long = if (remindLaterDuration > 0) remindLaterDuration
+        else remindLaterTimestamp - calendar.timeInMillis / 1000
+
+        // if fire date is greater than 0 then we want to schedule a reminder notification.
+        if (secondsUntilFireDate <= 0) {
+            Log.trace(
+                LOG_TAG,
+                SELF_TAG,
+                "Remind later date is before the current date. Will not reschedule the" +
+                    " notification.",
+                secondsUntilFireDate
+            )
+            // cancel the displayed notification
+            notificationManager.cancel(tag.hashCode())
+            return
+        }
+        Log.trace(
+            LOG_TAG,
+            SELF_TAG,
+            "Remind later pressed, will reschedule the notification to be displayed %d" +
+                " seconds from now",
+            secondsUntilFireDate
+        )
+        calendar.add(Calendar.SECOND, secondsUntilFireDate.toInt())
+
+        // schedule a pending intent to be broadcast at the specified timestamp
+        val scheduledIntent = Intent(PushTemplateConstants.IntentActions.SCHEDULED_NOTIFICATION_BROADCAST)
+        broadcastReceiverClass.let {
+            scheduledIntent.setClass(context.applicationContext, broadcastReceiverClass)
+        }
+        scheduledIntent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
+        scheduledIntent.putExtras(intentExtras)
+
+        val pendingIntent: PendingIntent = PendingIntent.getBroadcast(
+            context,
+            0,
+            scheduledIntent,
+            PendingIntent.FLAG_MUTABLE or PendingIntent.FLAG_UPDATE_CURRENT
+        )
+
+        val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager?
+        if (alarmManager != null) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && alarmManager.canScheduleExactAlarms()) {
+                Log.trace(
+                    LOG_TAG,
+                    SELF_TAG,
+                    "Exact alarms are permitted, scheduling an exact alarm for the current notification."
+                )
+                alarmManager.setExactAndAllowWhileIdle(
+                    AlarmManager.RTC_WAKEUP, calendar.timeInMillis, pendingIntent
+                )
+            } else {
+                // schedule an inexact alarm for the current notification
+                Log.trace(
+                    LOG_TAG,
+                    SELF_TAG,
+                    "Exact alarms are not permitted, scheduling an inexact alarm for the current notification."
+                )
+                alarmManager[AlarmManager.RTC_WAKEUP, calendar.timeInMillis] =
+                    pendingIntent
+            }
+            // cancel the displayed notification
+            notificationManager.cancel(tag.hashCode())
```

Comment:
thinking about this a bit more, i feel that that the notification should be canceled as well if no alarm manager is available. otherwise the notification would appear to be unresponsive when remind later is pressed. what do you think?

Reply:
Changed this logic to make sure notification is always cancelled in case of errors